File: 55. Jump Game/Solution.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        high, curr, n = 0, 0, len(nums)
        while curr <= high:
            high = max(high, curr + nums[curr])
            curr += 1
            if high >= n - 1: return True
        return False

File: 55. Jump Game/Solution.java
class Solution {
    public boolean canJump(int[] nums) {
        int high = 0, curr = 0, n = nums.length;
        while (curr <= high) {
            high = Math.max(high, curr + nums[curr]);
            curr++;
            if (high >= n - 1) return true;
        }
        return false;
    }
}
